### DISCUSSION TITLE
Is there a way to access the DisplayText on the selected item on ContentFieldPicker using liquid

### DISCUSSION THREAD

**Discussion by MikeAlhayek** [OrchardCMS/OrchardCore#11659] [Q&A] [answered]
Is there a way to access the DisplayText on the selected item on ContentFieldPicker using liquid?

In other words, I have a customer `ContentPickerField` on `X` content-type, and I want to use the selected customer name to generate a title for my `X` content when created a content item of `X`.

If this is not possible, alternately I would be able to achieve the same this using vanilla JavaScript if I am able to listen to the change event of the `vue-multieselect` element. What is the right way to use vanilla JS to listen for a value change on a `ContentPickerField` so I can get the displaytext and set it to the title manually?

**Comment by jtkech** [comment]
Here an example of the usage of the Content.ContentItemId liquid accessor.

        public const string CreditTitlePattern =
    @"{% assign creditPart = Model.ContentItem.Content.CreditPart %}
    {% assign agentId = creditPart.Agent.ContentItemIds[0] %}
    {% assign roleId = creditPart.Role.ContentItemIds[0] %}
    {% assign agent = Content.ContentItemId[agentId] | display_text %}
    {% assign role = Content.ContentItemId[roleId] | display_text %}
    {{ agent | append: ' - ' | append: role }}";

**Reply by MikeAlhayek** [reply]
@jtkech Thank for the info. I am still unable to get the displayText of the selected user.

Here is what I tried to generate the title on my opportunity content-type.
```
{% assign customerId = Model.Content.RequiredCustomerPart.Customer.ContentItemIds[0] %}
{{ Model.Content.ContentItemId[customerId] | display_text | append: ' - Opportunity' }}
```

The generated title is `- Opportunity` so it does not get the Customer's display text

Here is my content part

    public class RequiredCustomerPart : ContentPart
    {
        public ContentPickerField Customer { get; set; }
    }

I am not sure in your code example where `Content.ContentItemId[roleId]` is coming from? Does the `Model.Content contains a dictionary of the content items by id?


I also tried this

```
{% assign customerId = Model.Content.RequiredCustomerPart.Customer.ContentItemIds[0] %}
{{ Model.Content.RequiredCustomerPart.Customer[customerId] | display_text | append: ' - Opportunity' }}
```

**Comment by jtkech** [comment, marked as answer]
ContentItemId is a property accessor of a pseudo Content object.

    o.Scope.SetValue("Content", new ObjectValue(new LiquidContentAccessor()));
    o.MemberAccessStrategy.Register<LiquidContentAccessor, LiquidPropertyAccessor>("ContentItemId", ... =>
    {

So remove `Model.` from `Model.Content.ContentItemId[customerId]`

Just keep `Content.ContentItemId[customerId]`

**Reply by MikeAlhayek** [reply]
@jtkech thanks again. That did not worth either
This is what I have 
```
{% assign customerId = Model.Content.RequiredCustomerPart.Customer.ContentItemIds[0] %}
{{ Content.ContentItemId[customerId] | display_text | append: ' - Opportunity' }}
```

![image](https://user-images.githubusercontent.com/24724371/167315962-bc6cc489-5198-44c3-9db6-869440ae7553.png)

**Reply by jtkech** [reply]
Hmm, this is what I do

    {% assign creditPart = Model.ContentItem.Content.CreditPart %}
    {% assign agentId = creditPart.Agent.ContentItemIds[0] %}

So in your case you would have to do

    {% assign customerId = Model.ContentItem.Content.RequiredCustomerPart.Customer.ContentItemIds[0] %}

**Reply by MikeAlhayek** [reply]
That did it! thank you so much
